mkosi/__init__.py

```diff
@@ -789,23 +789,27 @@ def sign_efi_binary(context: Context, input: Path, output: Path) -> None:
         find_binary("sbsign", root=context.config.tools()) is not None
     ):
         with open(output, "wb") as f:
-            run(
-                [
+            cmd = [
                     "sbsign",
-                    "--key", context.config.secure_boot_key,
                     "--cert", context.config.secure_boot_certificate,
                     "--output", "/dev/stdout",
-                    input,
-                ],
-                stdout=f,
-                sandbox=context.sandbox(
-                    options=[
-                        "--ro-bind", context.config.secure_boot_key, context.config.secure_boot_key,
+                ]
+            options = [
                         "--ro-bind", context.config.secure_boot_certificate, context.config.secure_boot_certificate,
                         "--ro-bind", input, input,
                     ]
-                ),
-            )
+            if context.config.secure_boot_key_uri:
+                cmd += [
+                        "--engine", context.config.secure_boot_key_uri[:context.config.secure_boot_key_uri.find(":")],
```

Comment:
```suggestion
                        "--engine", context.config.secure_boot_key_uri.split(":")[0],
```
though I'd maybe prefer splitting that beforehand and having some handling if there is no colon in there.